python/lsst/ts/logging_and_reporting/web_app/services/rubin_nights_service.py

```diff
@@ -148,10 +121,94 @@ def get_open_close_dome(
+        dome_open = get_dome_open_close(day_min, day_max, clients["efd"])
         return dome_open
     except Exception as e:
         logger.error(f"Error getting open/close dome times from rubin_nights through EFD: {e}", exc_info=True)
         return pd.DataFrame()
+
+
+def get_context_feed(
+    dayobs_start: int,
+    dayobs_end: int,
+    auth_token: str = None,
+) -> list:
+    """
+    Retrieve consolidated context feed messages for a given range of
+    observation days.
+
+    This function queries the Rubin Observatory services via the
+    consolidated ScriptQueue messages, within the specified dayObs
+    range. It processes the results into JSON-safe records, ensuring
+    special float values (NaN, Infinity) are handled correctly for
+    serialization. Only the relevant context feed columns are retained
+    from the raw data.
+
+    Parameters
+    ----------
+    dayobs_start : int
+        The starting observation day (as an integer, e.g., YYYYMMDD).
+    dayobs_end : int
+        The ending observation day (as an integer, e.g., YYYYMMDD).
+    auth_token : str, optional
+        Authentication token used when connecting to external
+        services.
+
+    Returns
+    -------
+    list
+        A list containing two elements:
+        - records : list of dict
+            JSON-safe context feed records, one per row.
+        - cols : list of str
+            The column names included in the context feed.
+        Returns an empty list if an error occurs.
+    """
+    logger.info(
+        f"Getting context feed data for start: {dayobs_start}, end: {dayobs_end}"
+    )
+    try:
+        # Get connections to rubin_nights services
+        endpoints = get_clients(auth_token=auth_token)
+
+        # Convert dayobs_start and dayobs_end to t_start and t_end
+        t_start = Time(f"{rn_dayobs.day_obs_int_to_str(dayobs_start)}T12:00:00", format='isot', scale='utc')
+        t_end = Time(
+            f"{rn_dayobs.day_obs_int_to_str(dayobs_end)}T12:00:00",
+            format="isot",
+            scale="utc",
+        ) + TimeDelta(1, format="jd")
+
+        # Returns pandas dataframe and list
+        df, cols = get_consolidated_messages(t_start, t_end, endpoints)
+
+        # For now, we drop all data not in columns=cols.
+        # The dropped columns contain the raw data.
+
+        # If we end up needing the raw data, we will need to handle
+        # the non-serialisable types or bypass the default JSON
+        # encoding and return a Response directly, e.g.
+        # records = df.to_dict(orient="records") # list of row dicts
+        # json_ready = jsonable_encoder(records) # fixes problem types
+        # return JSONResponse(content=json_ready)
+
+        # Discard all columns that are not listed in cols
+        df_cols_only = df[df.columns.intersection(cols)]
+
+        # Convert special floats (nans and infs) to strings
+        # This ensures that JSON serialisation does not fail
```

Comment:
I suggest to try consolidating these different comments into the docstrings of the function.